### DISCUSSION TITLE
Switch to application instead of click here to edit

### DISCUSSION THREAD

**Discussion by RickSeiden** [arnoudkooi/SN-Utils#416] [Ideas]
If I'm in one application, say Employee Center, and I end up in a record in another application, say Employee Center Pro, I'll get a message at the top of the screen that says something along the lines of, "This record is in Employee Center Pro but the current application is Employee Center.  To edit this record, click here."

If you click the link, edit the record and save your changes, you will capture a record from the second application in an update set for the first application.  When you go to preview that in another instance, it will fail and you won't be able to commit it with the record from the second application in it.

It would be incredibly helpful if "To edit this record, click here" became, "To switch to this application, click here". Then the system would automatically give you a message that your update set has changed (because you switched applications), and you wouldn't capture a record from one application in an update set for another application.

**Comment by arnoudkooi** [comment]
I like the idea, I'll check!

**Comment by arnoudkooi** [comment]
In version 7.0.7.0 the following link is added:

![Google Chrome20230718 - 1935](https://github.com/arnoudkooi/ServiceNow-Utils/assets/840781/918224ff-c36e-46c7-8f6e-8b5a68a8f852)

I didn't want to override existing behavior but add it as an extra option.
Hope this helps!

**Reply by SxMShaDoW** [reply]
I wonder if it's possible to take it a step further and also show a few open update sets names in that scope next to "click here" and when clicked it will go to that scope and switch to the correct update set for you.

**Reply by mgrak** [reply]
You could run a async script that scans sys_update_xml  for that scope and shows the last 3 update sets that have been modified by sys_updated_on so you can quickly choose, that would be nice!

**Reply by arnoudkooi** [reply]
Dont want to add too much UI stuff... I could check to add a link that calls a variation of the /su slash command

**Comment by RickSeiden** [comment]
That will be incredibly helpful!  I can't wait to see it!

**Comment by RickSeiden** [comment]
Thank you so much. I shared this discussion with some folks at work, and
the feedback was very positive!

**Comment by mgrak** [comment]
You could run a async script that scans sys_update_xml  for that scope and shows the last 3 update sets that have been modified by sys_updated_on so you can quickly choose, that would be nice!

    On Tuesday, July 18, 2023 at 11:36:07 PM UTC, Dorian ***@***.***> wrote:  
 
 


I wonder if it's possible to take it a step further and also show a few open update sets names in that scope next to "click here" and when clicked it will go to that scope and switch to the correct update set for you.

**Comment by RickSeiden** [comment]
It's live and it is GLORIOUS!  It's exactly what I was envisioning.  Thank you so much! I know it seems like a small change, but this really is going to make dealing with scoped applications so much easier!

I made a one time US$10 donation to your Patreon account as a small thank you!

**Reply by arnoudkooi** [reply]
Thanks, Rick, appreciate it and glad you love it :)
